pkg/liqoctl/rest/configuration/create.go

```diff
@@ -0,0 +1,190 @@
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/printers"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	networkingv1alpha1 "github.com/liqotech/liqo/apis/networking/v1alpha1"
+	liqoconsts "github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/liqoctl/completion"
+	"github.com/liqotech/liqo/pkg/liqoctl/output"
+	"github.com/liqotech/liqo/pkg/liqoctl/rest"
+	"github.com/liqotech/liqo/pkg/utils/args"
+)
+
+const liqoctlCreateConfigurationLongHelp = `Create a Configuration.
+
+The Configuration resource is used to represent a remote cluster network configuration.
+
+Examples:
+  $ {{ .Executable }} create configuration my-cluster --cluster-id my-cluster-id \
+  --pod-cidr 10.0.0.0/16 --external-cidr 10.10.0.0/16`
+
+// Create creates a Configuration.
+func (o *Options) Create(ctx context.Context, options *rest.CreateOptions) *cobra.Command {
+	outputFormat := args.NewEnum([]string{"json", "yaml"}, "")
+
+	o.createOptions = options
+
+	cmd := &cobra.Command{
+		Use:     "configuration",
+		Aliases: []string{"config", "configurations"},
+		Short:   "Create a Configuration",
+		Long:    liqoctlCreateConfigurationLongHelp,
+		Args:    cobra.ExactArgs(1),
+
+		PreRun: func(cmd *cobra.Command, args []string) {
+			options.OutputFormat = outputFormat.Value
+			options.Name = args[0]
+			o.createOptions = options
+		},
+
+		Run: func(cmd *cobra.Command, args []string) {
+			output.ExitOnErr(o.handleCreate(ctx))
+		},
+	}
+
+	cmd.Flags().VarP(outputFormat, "output", "o",
+		"Output the resulting VirtualNode resource, instead of applying it. Supported formats: json, yaml")
+
+	cmd.Flags().StringVar(&o.ClusterID, "cluster-id", "", "The cluster ID of the remote cluster")
+	cmd.Flags().Var(&o.PodCIDR, "pod-cidr", "The pod CIDR of the remote cluster")
+	cmd.Flags().Var(&o.ExternalCIDR, "external-cidr", "The external CIDR of the remote cluster")
+	cmd.Flags().BoolVar(&o.Wait, "wait", false, "Wait for the Configuration to be ready")
+
+	runtime.Must(cmd.MarkFlagRequired("cluster-id"))
+	runtime.Must(cmd.MarkFlagRequired("pod-cidr"))
+	runtime.Must(cmd.MarkFlagRequired("external-cidr"))
+
+	runtime.Must(cmd.RegisterFlagCompletionFunc("output", completion.Enumeration(outputFormat.Allowed)))
+	runtime.Must(cmd.RegisterFlagCompletionFunc("cluster-id", completion.ClusterIDs(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+
+	return cmd
+}
+
+func (o *Options) handleCreate(ctx context.Context) error {
+	opts := o.createOptions
+	if opts.OutputFormat != "" {
+		opts.Printer.CheckErr(o.output(nil))
+		return nil
+	}
+
+	s := opts.Printer.StartSpinner("Creating configuration")
+
+	conf := forgeConfiguration(o.createOptions.Name, o.createOptions.Namespace,
+		o.ClusterID, o.PodCIDR.String(), o.ExternalCIDR.String())
```

Comment:
Is possible to move this before line 93 and pass conf to o.output? I'm sorry if I'm losing some details. In this way you can avoid to call another forgeConfiguration inside the printer.